Stability checker does not clean testing pods

In some cases, stability checker does not delete testing pods. Next run won't start because of errors, see logs:
```
{"level":"error","log":{"message":"----------------------------\n- Testing Kyma...\n----------------------------\n- Testing Core components...\nRUNNING: test-core-kubeless\nFAILED: test-core-kubeless, run `kubectl logs test-core-kubeless --namespace kyma-system` for more info\nRUNNING: test-core-core-acceptance\nPASSED: test-core-core-acceptance\nRUNNING: test-api-controller-acceptance\nPASSED: test-api-controller-acceptance\nRUNNING: connector-service-tests\nPASSED: connector-service-tests\nRUNNING: test-core-environments\nPASSED: test-core-environments\nRUNNING: test-core-core-ui-api-acceptance\nPASSED: test-core-core-ui-api-acceptance\nRUNNING: metadata-service-tests\nPASSED: metadata-service-tests\nRUNNING: test-core-monitoring\nPASSED: test-core-monitoring\nRUNNING: test-core-event-bus-tester\nPASSED: test-core-event-bus-tester\nError: 1 test(s) failed\n\u001b[0m\u001b[1mAll test pods should be terminated. Checking...\u001b[0m\n\u001b[32m\u001b[1mOK\u001b[0m\n\u001b[0m\u001b[1mTesting 'connector-service-tests'\u001b[0m\nTest of 'connector-service-tests' was successful\nLogs are not displayed after success\n\u001b[0m\u001b[1mEnd of testing 'connector-service-tests'\n\u001b[0m\n\u001b[0m\u001b[1mTesting 'test-api-controller-acceptance'\u001b[0m\nTest of 'test-api-controller-acceptance' was successful\nLogs are not displayed after success\n\u001b[0m\u001b[1mEnd of testing 'test-api-controller-acceptance'\n\u001b[0m\n\u001b[0m\u001b[1mTesting 'test-core-core-acceptance'\u001b[0m\nTest of 'test-core-core-acceptance' was successful\nLogs are not displayed after success\n\u001b[0m\u001b[1mEnd of testing 'test-core-core-acceptance'\n\u001b[0m\n\u001b[0m\u001b[1mTesting 'test-core-core-ui-api-acceptance'\u001b[0m\nTest of 'test-core-core-ui-api-acceptance' was successful\nLogs are not displayed after success\n\u001b[0m\u001b[1mEnd of testing 'test-core-core-ui-api-acceptance'\n\u001b[0m\n\u001b[0m\u001b[1mTesting 'test-core-environments'\u001b[0m\nTest of 'test-core-environments' was successful\nLogs are not displayed after success\n\u001b[0m\u001b[1mEnd of testing 'test-core-environments'\n\u001b[0m\n\u001b[0m\u001b[1mTesting 'test-core-event-bus-tester'\u001b[0m\nTest of 'test-core-event-bus-tester' was successful\nLogs are not displayed after success\n\u001b[0m\u001b[1mEnd of testing 'test-core-event-bus-tester'\n\u001b[0m\n\u001b[0m\u001b[1mTesting 'test-core-kubeless'\u001b[0m\n\u001b[31m'test-core-kubeless' has Failed status\u001b[0m\n\u001b[0m\u001b[1mFetching logs from 'test-core-kubeless'\u001b[0m\n","test-run-id":"5d11d1e8-1471-4dae-9543-102c30f016e1","time":"2018-09-06T06:20:07.063Z","type":"test-output"}}
{"level":"info","log":{"message":"Start test","test-run-id":"f8e76949-726a-415d-81d4-9d63e6d5d594","time":"2018-09-06T06:25:07.074Z"}}
{"level":"error","log":{"message":"Test end with error [start time: 2018-09-06 06:25:07.075140359 +0000 UTC m=+21306.372387583, duration: 9.035233494s]","test-run-id":"f8e76949-726a-415d-81d4-9d63e6d5d594","time":"2018-09-06T06:25:16.11Z"}}
{"level":"error","log":{"message":"----------------------------\n- Testing Kyma...\n----------------------------\n- Testing Core components...\nRUNNING: test-core-kubeless\nERROR: pods \"test-core-kubeless\" already exists\nRUNNING: test-core-core-acceptance\nERROR: pods \"test-core-core-acceptance\" already exists\nRUNNING: test-api-controller-acceptance\nERROR: pods \"test-api-controller-acceptance\" already exists\nRUNNING: connector-service-tests\nERROR: pods \"connector-service-tests\" already exists\nRUNNING: test-core-environments\nERROR: pods \"test-core-environments\" already exists\nRUNNING: test-core-core-ui-api-acceptance\nERROR: pods \"test-core-core-ui-api-acceptance\" already exists\nRUNNING: metadata-service-tests\nERROR: pods \"metadata-service-tests\" already exists\nRUNNING: test-core-monitoring\nERROR: pods \"test-core-monitoring\" already exists\nRUNNING: test-core-event-bus-tester\nERROR: pods \"test-core-event-bus-tester\" already exists\nError: 9 test(s) failed\n\u001b[0m\u001b[1mAll test pods should be terminated. Checking...\u001b[0m\n\u001b[32m\u001b[1mOK\u001b[0m\n\u001b[0m\u001b[1mTesting 'connector-service-tests'\u001b[0m\nTest of 'connector-service-tests' was successful\nLogs are not displayed after success\n\u001b[0m\u001b[1mEnd of testing 'connector-service-tests'\n\u001b[0m\n\u001b[0m\u001b[1mTesting 'test-api-controller-acceptance'\u001b[0m\nTest of 'test-api-controller-acceptance' was successful\nLogs are not displayed after success\n\u001b[0m\u001b[1mEnd of testing 'test-api-controller-acceptance'\n\u001b[0m\n\u001b[0m\u001b[1mTesting 'test-core-core-acceptance'\u001b[0m\nTest of 'test-core-core-acceptance' was successful\nLogs are not displayed after success\n\u001b[0m\u001b[1mEnd of testing 'test-core-core-acceptance'\n\u001b[0m\n\u001b[0m\u001b[1mTesting 'test-core-core-ui-api-acceptance'\u001b[0m\nTest of 'test-core-core-ui-api-acceptance' was successful\nLogs are not displayed after success\n\u001b[0m\u001b[1mEnd of testing 'test-core-core-ui-api-acceptance'\n\u001b[0m\n\u001b[0m\u001b[1mTesting 'test-core-environments'\u001b[0m\nTest of 'test-core-environments' was successful\nLogs are not displayed after success\n\u001b[0m\u001b[1mEnd of testing 'test-core-environments'\n\u001b[0m\n\u001b[0m\u001b[1mTesting 'test-core-event-bus-tester'\u001b[0m\nTest of 'test-core-event-bus-tester' was successful\nLogs are not displayed after success\n\u001b[0m\u001b[1mEnd of testing 'test-core-event-bus-tester'\n\u001b[0m\n\u001b[0m\u001b[1mTesting 'test-core-kubeless'\u001b[0m\n\u001b[31m'test-core-kubeless' has Failed status\u001b[0m\n\u001b[0m\u001b[1mFetching logs from 'test-core-kubeless'\u001b[0m\n","test-run-id":"f8e76949-726a-415d-81d4-9d63e6d5d594","time":"2018-09-06T06:25:16.11Z","type":"test-output"}}
{"level":"info","log":{"message":"Start test","test-run-id":"20062335-83d9-44f4-9967-1fca7adb1b60","time":"2018-09-06T06:30:16.122Z"}}
{"level":"error","log":{"message":"Test end with error [start time: 2018-09-06 06:30:16.122119249 +0000 UTC m=+21615.419366473, duration: 12.180667833s]","test-run-id":"20062335-83d9-44f4-9967-1fca7adb1b60","time":"2018-09-06T06:30:28.302Z"}}```

Issues to create:
- run Dex tests also on cluster. 
Currently, in the "dex_test.go" there is a following condition:
```
 !testRunningInLocalEnv {
		t.Skip()
}
```
This cause that dex tests are not running on clusters. What is even more dangerous, it is not obvious that we skip this test. IMO tests should be written in the way that it should be runnable locally and in the cluster. 

2. UI-API-Layer tests fails on the cluster. 
For example:
```
--- FAIL: TestLimitRangeQuery (0.91s)
 	limitrange_test.go:32: Creating namespace...
 	limitrange_test.go:53: Querying for Limit Ranges...
 	assertions.go:247:                            	Error Trace: 	limitrange_test.go:56
 	 	 	Error:       	Received unexpected error:
 	 	 	             	Post https://ui-api.nightly.cluster.kyma.cx/graphql: Request canceled.: Login - response error: '400 Bad Request' - \u003c!DOCTYPE html\u003e
 	 	 	             	\u003chtml\u003e
 	 	 	             	  \u003chead\u003e
 	 	 	             	    \u003cmeta charset=\"utf-8\"\u003e
 	 	 	             	    \u003cmeta http-equiv=\"X-UA-Compatible\" content=\"IE=edge,chrome=1\"\u003e
 	 	 	             	    \u003ctitle\u003edex\u003c/title\u003e
 	 	 	             	    \u003cmeta name=\"viewport\" content=\"width=device-width, initial-scale=1.0\"\u003e
 	 	 	             	    \u003clink href=\"https://dex.nightly.cluster.kyma.cx/static/main.css\" rel=\"stylesheet\"\u003e
 	 	 	             	    \u003clink href=\"https://dex.nightly.cluster.kyma.cx/theme/styles.css\" rel=\"stylesheet\"\u003e
 	 	 	             	    \u003clink rel=\"icon\" href=\"https://dex.nightly.cluster.kyma.cx/theme/favicon.png\"\u003e
 	 	 	             	  \u003c/head\u003e
 	 	 	             	
 	 	 	             	  \u003cbody class=\"theme-body\"\u003e
 	 	 	             	    \u003cdiv class=\"theme-navbar\"\u003e
 	 	 	             	      \u003cdiv class=\"theme-navbar__logo-wrap\"\u003e
 	 	 	             	        \u003cimg class=\"theme-navbar__logo\" src=\"https://dex.nightly.cluster.kyma.cx/theme/logo.png\"\u003e
 	 	 	             	      \u003c/div\u003e
 	 	 	             	    \u003c/div\u003e
 	 	 	             	
 	 	 	             	    \u003cdiv class=\"dex-container\"\u003e
 	 	 	             	
 	 	 	             	
 	 	 	             	
 	 	 	             	\u003cdiv class=\"theme-panel\"\u003e
 	 	 	             	  \u003ch2 class=\"theme-heading\"\u003eBad Request\u003c/h2\u003e
 	 	 	             	  \u003cp\u003eRequested resource does not exist.\u003c/p\u003e
 	 	 	             	\u003c/div\u003e
 	 	 	             	
 	 	 	             	    \u003c/div\u003e
 	 	 	             	  \u003c/body\u003e
 	 	 	             	\u003c/html\u003e
 	 	 	             	
 	 	 	Test:        	TestLimitRangeQuery
 	limitrange_test.go:37: Deleting namespace...
```

--- a/installation/scripts/testing.sh
+++ b/installation/scripts/testing.sh
@@ -116,7 +116,7 @@ function cleanupHelmTestPods() {
       log "FAILED cleaning test pods.\n" red
       return 1
     fi
-    log "End of cleaning test pods.\n" nc bold
+    log "Success cleaning test pods.\n" nc bold
 }
 
 function waitForTestPodsTermination() {

--- a/installation/scripts/testing.sh
+++ b/installation/scripts/testing.sh
@@ -110,6 +110,12 @@ function cleanupHelmTestPods() {
 
     log "\nCleaning up helm test pods" nc bold
     kubectl delete pod -n ${namespace} -l helm-chart-test=true
+    deleteErr=$?
+    if [ ${deleteErr} -ne 0 ]
+    then
+      log "FAILED cleaning test pods.\n" red
+      return 1
+    fi
     log "End of cleaning test pods.\n" nc bold
 }
 
@@ -146,8 +152,9 @@ function checkAndCleanupTest() {
     checkTestPodLabelErr=$?
 
     cleanupHelmTestPods ${namespace}
+    cleanupErr=$?
 
-    if [ ${checkTestPodTerminatedErr} -ne 0 ] || [ ${checkTestPodLabelErr} -ne 0 ]
+    if [ ${checkTestPodTerminatedErr} -ne 0 ] || [ ${checkTestPodLabelErr} -ne 0 ] | [ ${cleanupErr} -ne 0 ]
     then
         return 1
     fi
@@ -193,8 +200,9 @@ testCheckIstio=$?
 
 echo "- Testing Remote Environments"
 helm test ec-default
+ecTestErr=$?
 helm test hmc-default
-gatewayTestErr=$?
+hmcTestErr=$?
 
 checkAndCleanupTest kyma-integration
 testCheckGateway=$?
@@ -202,7 +210,7 @@ testCheckGateway=$?
 printImagesWithLatestTag
 latestTagsErr=$?
 
-if [ ${latestTagsErr} -ne 0 ] || [ ${coreTestErr} -ne 0 ]  || [ ${istioTestErr} -ne 0 ] || [ ${gatewayTestErr} -ne 0 ]
+if [ ${latestTagsErr} -ne 0 ] || [ ${coreTestErr} -ne 0 ]  || [ ${istioTestErr} -ne 0 ] || [ ${ecTestErr} -ne 0 ] || [ ${hmcTestErr} -ne 0 ]
 then
     exit 1
 else

